<?xml version="1.0"?>

<launch>
  <arg name="namespace" default="" />
  <arg name="ideal_dist" default="10" />
  <arg name="target_vel"  default="1.0" />
  <arg name="vel_coef" default="1.0" />
  <arg name="dist_coef"   default="2.5" />
  <arg name="rate"   default="12.0" />

  <group ns="/$(arg namespace)">
    <node name="reward_follow_shore" pkg="reward_generator" type="shore_follower" args="" output="screen">
      <remap from="~/laser"     to="front/scan_360" />
      <remap from="~/odom"      to="pose_gt" />
      <remap from="~/done"      to="agent/is_done" />
      <remap from="~/refresh"   to="server/episode_manager" />
      
      <param name="dist_to_shore"          type="double"  value="$(arg ideal_dist)" />
      <param name="rate"      	 	   type="double"  value="$(arg rate)" />
      <param name="target_vel"             type="double"  value="$(arg target_vel)" />
      <param name="dist_coeff"             type="double"  value="$(arg dist_coef)" />
      <param name="vel_coeff"              type="double"  value="$(arg vel_coef)" />
      <param name="laser_frame"            type="string"  value="front_laser" />
      <param name="safety_area_cost"       type="int"     value="500" />
      <param name="max_gap"                type="double"  value="0.5" />
      <param name="publish_occupancy_grid" type="bool"    value="true" />
      <param name="blur_kernel_size"       type="double"  value="1.5" />
    </node>
  </group>
</launch>
